src/player/player_update.cpp

```diff
@@ -2218,28 +2224,43 @@ void UpdatePlayer()
                 // Walk offscreen exit
                 if(!hBoundsHandled && OffScreenExit[Player[A].Section])
                 {
+                    bool offScreenExit = false;
                     if(Player[A].Location.X + Player[A].Location.Width < level[Player[A].Section].X)
                     {
-                        LevelBeatCode = 3;
-                        EndLevel = true;
+                        offScreenExit = true;
                         for(B = 1; B <= numPlayers; B++)
                             Player[B].TailCount = 0;
-                        LevelMacro = LEVELMACRO_OFF;
-                        LevelMacroCounter = 0;
-                        frmMain.setTargetTexture();
-                        frmMain.clearBuffer();
-                        frmMain.repaint();
                     }
                     else if(Player[A].Location.X > level[Player[A].Section].Width)
+                    {
+                        offScreenExit = true;
+                    }
+
+                    if(offScreenExit)
                     {
                         LevelBeatCode = 3;
                         EndLevel = true;
                         LevelMacro = LEVELMACRO_OFF;
                         LevelMacroCounter = 0;
-                        frmMain.setTargetTexture();
-                        frmMain.clearBuffer();
-                        frmMain.repaint();
+                        g_levelScreenFader.setupFader(4, 0, 65, ScreenFader::S_FADE);
+
+                        while(!g_levelScreenFader.m_full && GameIsActive)
```

Comment:
Maybe blocks like this could be placed in a separate "WaitForFader" method? I saw this pattern repeated a lot and even if the same one is not used multiple times in code, breaking this out into its own method (even `static inline`'d) could help readability some.

Reply:
Good point

Reply:
Really, could be better, also, you just now gave me an idea to solve the "everyone dead" task! So, I made this call and I re-used it at "EveryonesDead()" call

Reply:
Done, should be better now :wink: 
Also, fade now works when all players have gone